// @nexeraprotocol/identity-schemas/identity/dist/nexeraprotocol-identity-schemas-identity.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var kyc_schema = require('../../dist/kyc.schema-42cb9d1a.cjs.dev.js');
require('zod');



exports.ARRAY_OPERATORS = kyc_schema.ARRAY_OPERATORS;
exports.AVAILABLE_FLOWS = kyc_schema.AVAILABLE_FLOWS;
exports.AllCredentialAttributes = kyc_schema.AllCredentialAttributes;
exports.AllCredentialValues = kyc_schema.AllCredentialValues;
exports.AllScenarioExecutionAuthorizationData = kyc_schema.AllScenarioExecutionAuthorizationData;
exports.ArrayOperator = kyc_schema.ArrayOperator;
exports.AuthQrCodeData = kyc_schema.AuthQrCodeData;
exports.AuthorizationRequestMessage = kyc_schema.AuthorizationRequestMessage;
exports.AuthorizationResponseMessage = kyc_schema.AuthorizationResponseMessage;
exports.AvailableFlow = kyc_schema.AvailableFlow;
exports.BOOLEAN_OPERATORS = kyc_schema.BOOLEAN_OPERATORS;
exports.BooleanOperator = kyc_schema.BooleanOperator;
exports.ChallengeQueryOperator = kyc_schema.ChallengeQueryOperator;
exports.ChallengeQueryOperatorToOperator = kyc_schema.ChallengeQueryOperatorToOperator;
exports.ChallengeQueryOperators = kyc_schema.ChallengeQueryOperators;
exports.ChallengeQuerySchema = kyc_schema.ChallengeQuerySchema;
exports.CloseScreenNotification = kyc_schema.CloseScreenNotification;
exports.ComplianceImplementationStepsInput = kyc_schema.ComplianceImplementationStepsInput;
exports.CreateAuthRequestProps = kyc_schema.CreateAuthRequestProps;
exports.CreateQueryConfigurationInput = kyc_schema.CreateQueryConfigurationInput;
exports.CreateQueryConfigurationResponse = kyc_schema.CreateQueryConfigurationResponse;
exports.CreateZKProofRequestProps = kyc_schema.CreateZKProofRequestProps;
exports.CredentialQrCodeData = kyc_schema.CredentialQrCodeData;
exports.DATE_OPERATORS = kyc_schema.DATE_OPERATORS;
exports.DataAvailableOnStart = kyc_schema.DataAvailableOnStart;
exports.DateOperator = kyc_schema.DateOperator;
exports.DeleteQueryConfigurationInput = kyc_schema.DeleteQueryConfigurationInput;
exports.DeleteQueryConfigurationResponse = kyc_schema.DeleteQueryConfigurationResponse;
exports.ExecuteChallengeQueryConfig = kyc_schema.ExecuteChallengeQueryConfig;
exports.ExecuteChallengeQueryInput = kyc_schema.ExecuteChallengeQueryInput;
exports.ExecuteChallengeQueryResponse = kyc_schema.ExecuteChallengeQueryResponse;
exports.GetCredentialsRequest = kyc_schema.GetCredentialsRequest;
exports.GetCredentialsResponse = kyc_schema.GetCredentialsResponse;
exports.GetTxAuthDataSignatureResponse = kyc_schema.GetTxAuthDataSignatureResponse;
exports.GetTxAuthSigRequest = kyc_schema.GetTxAuthSigRequest;
exports.GetTxAuthSigResponse = kyc_schema.GetTxAuthSigResponse;
exports.HostMessage = kyc_schema.HostMessage;
exports.HostRequestMessage = kyc_schema.HostRequestMessage;
exports.HostResponseMessage = kyc_schema.HostResponseMessage;
exports.IDInformation = kyc_schema.IDInformation;
exports.IdentityMessage = kyc_schema.IdentityMessage;
exports.IdentityNotificationMessage = kyc_schema.IdentityNotificationMessage;
exports.IdentityRequestMessage = kyc_schema.IdentityRequestMessage;
exports.IdentityResponseMessage = kyc_schema.IdentityResponseMessage;
exports.InitialDataRequest = kyc_schema.InitialDataRequest;
exports.InitialDataResponse = kyc_schema.InitialDataResponse;
exports.IsVerifiedNotification = kyc_schema.IsVerifiedNotification;
exports.IsVerifiedRequest = kyc_schema.IsVerifiedRequest;
exports.IsVerifiedResponse = kyc_schema.IsVerifiedResponse;
exports.KycCompletionData = kyc_schema.KycCompletionData;
exports.KycCompletionNotification = kyc_schema.KycCompletionNotification;
exports.MediaType = kyc_schema.MediaType;
exports.MediaTypePID = kyc_schema.MediaTypePID;
exports.NUMERIC_OPERATORS = kyc_schema.NUMERIC_OPERATORS;
exports.NumericOperator = kyc_schema.NumericOperator;
exports.OPAChallengeQuery = kyc_schema.OPAChallengeQuery;
exports.OcvSdkInitialized = kyc_schema.OcvSdkInitialized;
exports.OffChainScenarioExecutionData = kyc_schema.OffChainScenarioExecutionData;
exports.OffChainZKP = kyc_schema.OffChainZKP;
exports.OffChainZKPRuleResult = kyc_schema.OffChainZKPRuleResult;
exports.OnChainRuleResult = kyc_schema.OnChainRuleResult;
exports.OnChainScenarioExecutionData = kyc_schema.OnChainScenarioExecutionData;
exports.OnChainZKP = kyc_schema.OnChainZKP;
exports.OnChainZKPFromNexeraID = kyc_schema.OnChainZKPFromNexeraID;
exports.OnChainZKPPayload = kyc_schema.OnChainZKPPayload;
exports.OnChainZKPScenarioFromCms = kyc_schema.OnChainZKPScenarioFromCms;
exports.OpaChallengeQueryResponse = kyc_schema.OpaChallengeQueryResponse;
exports.Operator = kyc_schema.Operator;
exports.OperatorToChallengeQueryOperator = kyc_schema.OperatorToChallengeQueryOperator;
exports.PartialQueryConfigSimplified = kyc_schema.PartialQueryConfigSimplified;
exports.PolygonIdInitialized = kyc_schema.PolygonIdInitialized;
exports.PolygonIdRequest = kyc_schema.PolygonIdRequest;
exports.PolygonIdRequestData = kyc_schema.PolygonIdRequestData;
exports.PolygonIdResponse = kyc_schema.PolygonIdResponse;
exports.PolygonIdResponseData = kyc_schema.PolygonIdResponseData;
exports.ProjectAuthorizationData = kyc_schema.ProjectAuthorizationData;
exports.ProofData = kyc_schema.ProofData;
exports.QrCodeLinkWithSchemaType = kyc_schema.QrCodeLinkWithSchemaType;
exports.QueryConfig = kyc_schema.QueryConfig;
exports.QueryConfigSimplified = kyc_schema.QueryConfigSimplified;
exports.QueryConfigSimplifiedParsed = kyc_schema.QueryConfigSimplifiedParsed;
exports.QueryConfiguration = kyc_schema.QueryConfiguration;
exports.QueryCredentialType = kyc_schema.QueryCredentialType;
exports.QueryCredentialTypes = kyc_schema.QueryCredentialTypes;
exports.RequiredDataRowSchema = kyc_schema.RequiredDataRowSchema;
exports.RequiredVerificationData = kyc_schema.RequiredVerificationData;
exports.RuleEngineResponse = kyc_schema.RuleEngineResponse;
exports.RuleEngineScenarioExecutionData = kyc_schema.RuleEngineScenarioExecutionData;
exports.RuleResultStatus = kyc_schema.RuleResultStatus;
exports.RuleResultStatusLabels = kyc_schema.RuleResultStatusLabels;
exports.RuleResultStatuses = kyc_schema.RuleResultStatuses;
exports.SCENARIO_AUTHORIZATION_STATUSES = kyc_schema.SCENARIO_AUTHORIZATION_STATUSES;
exports.STRING_OPERATORS = kyc_schema.STRING_OPERATORS;
exports.SUPPORTED_TYPES = kyc_schema.SUPPORTED_TYPES;
exports.ScenarioAuthorizationData = kyc_schema.ScenarioAuthorizationData;
exports.ScenarioAuthorizationStatus = kyc_schema.ScenarioAuthorizationStatus;
exports.ScenarioExecutionData = kyc_schema.ScenarioExecutionData;
exports.ScenarioExecutionNotification = kyc_schema.ScenarioExecutionNotification;
exports.ScenarioExecutionResponse = kyc_schema.ScenarioExecutionResponse;
exports.ScenarioStatus = kyc_schema.ScenarioStatus;
exports.ScenarioStatuses = kyc_schema.ScenarioStatuses;
exports.ScenarioType = kyc_schema.ScenarioType;
exports.ScenarioTypes = kyc_schema.ScenarioTypes;
exports.SdkVerificationOutput = kyc_schema.SdkVerificationOutput;
exports.SdkVerificationResponseSchema = kyc_schema.SdkVerificationResponseSchema;
exports.SendTransactionRequest = kyc_schema.SendTransactionRequest;
exports.SignatureRequest = kyc_schema.SignatureRequest;
exports.SignatureResponse = kyc_schema.SignatureResponse;
exports.SimplifiedCredential = kyc_schema.SimplifiedCredential;
exports.StartCompletedNotification = kyc_schema.StartCompletedNotification;
exports.StartFlowRequest = kyc_schema.StartFlowRequest;
exports.StringOperator = kyc_schema.StringOperator;
exports.SupportedType = kyc_schema.SupportedType;
exports.SupportedTypes = kyc_schema.SupportedTypes;
exports.TransactionData = kyc_schema.TransactionData;
exports.TransactionResponse = kyc_schema.TransactionResponse;
exports.TxAuthInput = kyc_schema.TxAuthInput;
exports.TxSignatureResponse = kyc_schema.TxSignatureResponse;
exports.UpdateQueryConfigurationInput = kyc_schema.UpdateQueryConfigurationInput;
exports.UpdateQueryConfigurationResponse = kyc_schema.UpdateQueryConfigurationResponse;
exports.ZKPOperator = kyc_schema.ZKPOperator;
exports.ZKPRequest = kyc_schema.ZKPRequest;
exports.ZKPRequestFromZKVerifier = kyc_schema.ZKPRequestFromZKVerifier;
exports.ZeroKnowledgeProofRequest = kyc_schema.ZeroKnowledgeProofRequest;
exports.ZeroKnowledgeProofResponse = kyc_schema.ZeroKnowledgeProofResponse;
exports.operatorDisplayMap = kyc_schema.operatorDisplayMap;
exports.parseIden3Message = kyc_schema.parseIden3Message;
exports.parseSessionIdFromUrl = kyc_schema.parseSessionIdFromUrl;
